lib/Conversion/TCFToLinalg/TCFToLinalg.cpp

```diff
@@ -34,12 +34,47 @@ static SmallVector<Value, 6> bypassResultShapes(Operation *op,
   }
   // TODO: This only supports the NCHW data format. Consider other formats and lower ranks.
   if (auto conv2dNCHW = dyn_cast<tcf::ConvNCHWOp>(op)) {
+    // TODO: Replace hard-coded stride/dilation/padding constant-ops.
```

Comment:
Maybe add a link to the PyTorch docs?

Reply:
Not sure about the PyTorch link since this op is out of PyTorch's scope. Also where's the best place to add a link? I'm thinking add a reference link to `include/npcomp/Dialect/TCF/TCFOps.td`. Do you know of a more framework-agnostic, maybe canonical, documentation for convolution operations? I'm looking around but may settle for PyTorch's docs.

Another topic related to the hard-coded stride/dilation/padding constant-ops, those are obviously placeholders and I noticed that the TCF->Linalg pass optimizes out the constant-ops, which makes sense. I don't know if it's possible to block optimization or how to declare those attributes as variables with default values.

Reply:
Linking to the PyTorch docs is fine. This is documentation for a mathematical operator which has specific deep learning semantics and details. I don't know anywhere that is a more straightforward and canonical place to link to.

As far as the hard-coded stride/dilation/padding -- I was actually expecting the pass to only create the simplified IR in the first place. But the way you are doing it is fine as well.